### ISSUE TITLE
Error loading plugin on Homebridge start

### ISSUE THREAD

**Issue by Tiger519** [leematt/homebridge-tplink-lightbulb#4]
I'm getting the following issue trying to run Homebridge after reinstalling the plugin:

ERROR LOADING PLUGIN homebridge-tplink-lightbulb:
[5/18/2017, 12:24:55 PM] SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:404:25)
    at Object.Module._extensions..js (module.js:432:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at Plugin.load (C:\Users\coolt\AppData\Roaming\npm\node_modules\homebridge\lib\plugin.js:65:22)
    at Server.<anonymous> (C:\Users\coolt\AppData\Roaming\npm\node_modules\homebridge\lib\server.js:132:14)
    at Array.forEach (native)

I previouly had this plugin installed and removed it when I was having connectivity issues. Running npm 4.5.0 and the only other accessory I'm using is Kevo smart locks (they work fine when I remove the platform definition from the config.json).

**Comment by Tiger519** [comment]
After some googling it appears this is because I'm running nodejs 4.5.0 and these functions are only supported in strict mode with that version. I'm guessing there are reasons not to use strict mode (I only dabble in programming), so the solution would appear to be to upgrade nodejs. What version are you running? The HomeBridge wiki says to use version 5, so I could upgrade to that but it appears the same functions required strict mode in that version.

Also, not sure if the 0.10.33 release will work with HomeBridge or solve the strict mode issue. Was considering running that version for AirSonos. I'm a bit new to this stuff, and trying to mess around with different versions worries me about screwing the whole thing up.

**Comment by leematt** [comment]
I've only tested the package with Node.js 6.x and 7.10. I've just added the `'use strict'` directive to tell older versions to run the package in strict mode and published `v1.0.6` to npm. Try it out and see if it'll run!

**Comment by Tiger519** [comment]
Thanks. I'll probably be able to check it out next week. I'll try running on 4.5 first and go from there. I assume you're running HomeBridge on 6.x/7.10 with no issue? I'll report back here for science!

**Comment by leematt** [comment]
Yep, I've been using about 8 Homebridge plugins on 6.x/7.10 without issue (well, there's always issues trying to use Homebridge/HomeKit, but nothing I've tied to Node.js). I haven't used the Kevo plugin though.

Edit: Actually, I've been using 6.6 in my production environment with most of the plugins. This plugin and a few others have been fine in 7.10.

**Comment by Tiger519** [comment]
Lightbulbs appear to be fine in 4.5.0 now! Thanks. And I think it may have been a coincidence that all of HomeBridge was crashing as I reported in issue #2. After I removed the lightbulb plugin HomeBridge ran fine for over a month, but my computer installed an update and restarted, and I was back having to restart HomeBridge every 2 minutes. Running in debug mode it seems that the issue is the TCP connection is closing, and the hap.tcp service actually stops being advertised. Sometimes restarting iTunes (on the same PC) actually seems to "wake up" the service, which is quite strange to me. Mostly just wanted to share that this is working now, though if you've worked through any similar TCP connection issues I'd welcome your input!
